learn-arai/app/hooks/useUser.ts

```diff
@@ -0,0 +1,53 @@
+'use client';
+
+import { useEffect, useState } from 'react';
+
+import { useLocalStorage } from './useLocalStorage';
+
+export type User = {
+    email: string;
+};
+
+export const useUser = () => {
+    const [user, setUser] = useState<User | null>(null);
+    const { setItem, removeItem, getItem } = useLocalStorage();
+
+    const addUser = (user: User) => {
+        setUser(user);
+        setItem('user', JSON.stringify(user));
+    };
+
+    const getUserFromLocalStorage = async () => {
+        let user = getItem('user');
+        if (!user) {
+            addUser(await fetchedUser());
+            user = getItem('user');
+        }
+
+        setUser(JSON.parse(user!));
+
+        // in case there is no user in local storage.
+    };
```

Comment:
Shouldn't useUser hook interact with useLocalStorage shouldn't it be useAuth hook?

Reply:
I don't understand this comment. You mean `getUserFromLocalStorage` should be in `useLocalStorage` or this one should be in `useAuth`.

Please explain more details.

Reply:
It should be in `useAuth` hooks,

moved this logic into `useEffect` of `useAuth` hook, and change it up a bit like
```
let user = getItem('user');
if(user) {
    checkSession();
    // (when session is expired, remove from user hook and localstorage hook)
    // return ...
}
// here you don't have to do anything, this case won't happen anyways
// (when client's data is missing, but session is valid)
```

Reply:
ok